Nos/Views/ReportMenu.swift

```diff
@@ -0,0 +1,143 @@
+                actions: { 
+                    Button(Localized.confirm.string) { 
+                        publishReport()
+                        if let author = reportedObject.author, !author.muted {
+                            showMuteDialog = true
+                        }
+                    }
+                    Button(Localized.cancel.string, role: .cancel) { 
+                        selectedCategory = nil
+                    }
+                },
+                message: {
+                    Localized.reportConfirmation.view
+                }
+            ) 
+            // Mute user menu
+            .alert(
+                Localized.muteUser.string, 
+                isPresented: $showMuteDialog,
+                actions: { 
+                    if let author = reportedObject.author {
+                        Button(Localized.yes.string) { 
+                            Task { await author.mute(context: viewContext) }
+                        }
+                        Button(Localized.no.string) {}
+                    }
+                },
+                message: {
+                    if let author = reportedObject.author {
+                        Localized.mutePrompt.view(["user": author.safeName])
+                    } else {
+                        Localized.error.view
+                    }
+                }
+            ) 
+    }
+    
+    /// Generates a series of Buttons for the given report categories.
+    func subCategoryButtons(for categories: [ReportCategory]) -> some View {
+        ForEach(categories) { subCategory in
+            Button(subCategory.displayName) { 
+                self.selectedCategory = subCategory
+                if subCategory.subCategories?.count ?? 0 == 0 {
+                    confirmReport = true
+                } else {
+                    Task { 
+                        self.isPresented = true
+                    }
+                }
+            }
+        }
+    }
+    
+    /// Publishes a report for the currently selected 
+    func publishReport() {
+        guard let keyPair = currentUser.keyPair,
+            let selectedCategory else {
+            Log.error("Cannot publish report - No signed in user")
+            return 
+        }
+        let event = JSONEvent(
+            pubKey: keyPair.publicKeyHex, 
+            kind: .label, 
+            tags: [
+                ["L", "MOD"],
+                ["l", selectedCategory.code, "MOD"],
+            ], 
+            content: ""
+        )
+        
+        Task {
+            do {
+                try await relayService.publishToAll(event: event, signingKey: keyPair, context: viewContext)
+                analytics.reported(reportedObject)
+            } catch {            
+                Log.error("Failed to publish report: \(error.localizedDescription)")
+            }
+        }
+    }
+}
+
+extension View {
+    func reportMenu(_ show: Binding<Bool>, reportedObject: ReportTarget) -> some View {
```

Comment:
This modifier needs documentation

Reply:
It could be a simple copy and paste of what you have above, so that it appears when you option-click the modifier outside this class.